sys/shell/shell.c

```diff
@@ -217,14 +225,34 @@ static void handle_input_line(const shell_command_t *command_list, char *line)
     }
 }
 
+/**
+ * Read a single line from standard input into a buffer.
+ *
+ * In addition to copying characters, this routine echoes the line back to
+ * stdout and also supports primitive line editing.
+ *
+ * If the input line is too long, the input will still be consumed until the end
+ * to prevent the next line from containing garbage.
+ *
+ * @param   buf     Buffer where the input will be placed.
+ * @param   size    Size of the buffer. The maximum line length will be one less
+ *                  than size, to accommodate for the null terminator.
+ *                  The minimum buffer size is 1.
+ *
+ * @return  READLINE_HASTEXT if the line was not empty
```

Comment:
I don't think we should do some non-standard error reporting here. -1 on EOF, 0 ... 2 for other errors? EOF defined in a different header, the rest in a local enum? That's at least confusing.

How about we change the signature of the function to "size_t".
Empty line returns "0". Line with text returns it's length.
For the error cases, we define an enum, or re-use something from errno. ENOBUFS, EBADFD  come to mind.